src/test/javascript/spec/integration/code-editor/code-editor-container.integration.spec.ts

```diff
@@ -226,52 +212,52 @@ describe('CodeEditorContainerIntegration', () => {
         containerFixture.detectChanges();
 
         // container
-        expect(container.commitState).to.equal(CommitState.CLEAN);
-        expect(container.editorState).to.equal(EditorState.CLEAN);
-        expect(container.buildOutput.isBuilding).to.be.false;
-        expect(container.unsavedFiles).to.be.empty;
+        expect(container.commitState).toEqual(CommitState.CLEAN);
+        expect(container.editorState).toEqual(EditorState.CLEAN);
+        expect(container.buildOutput.isBuilding).toBe(false);
+        expect(container.unsavedFiles).toBeEmpty;
 
         // file browser
-        expect(checkIfRepositoryIsCleanStub).to.have.been.calledOnce;
-        expect(getRepositoryContentStub).to.have.been.calledOnce;
-        expect(container.fileBrowser.errorFiles).to.deep.equal(extractedErrorFiles);
-        expect(container.fileBrowser.unsavedFiles).to.be.empty;
+        expect(checkIfRepositoryIsCleanStub).toHaveBeenCalledTimes(1);
+        expect(getRepositoryContentStub).toHaveBeenCalledTimes(1);
+        expect(container.fileBrowser.errorFiles).toEqual(extractedErrorFiles);
+        expect(container.fileBrowser.unsavedFiles).toBeEmpty();
 
         // ace editor
-        expect(container.aceEditor.isLoading).to.be.false;
-        expect(container.aceEditor.commitState).to.equal(CommitState.CLEAN);
+        expect(container.aceEditor.isLoading).toBe(false);
+        expect(container.aceEditor.commitState).toEqual(CommitState.CLEAN);
 
         // actions
-        expect(container.actions.commitState).to.equal(CommitState.CLEAN);
-        expect(container.actions.editorState).to.equal(EditorState.CLEAN);
-        expect(container.actions.isBuilding).to.be.false;
+        expect(container.actions.commitState).toEqual(CommitState.CLEAN);
+        expect(container.actions.editorState).toEqual(EditorState.CLEAN);
+        expect(container.actions.isBuilding).toBe(false);
 
         // status
-        expect(container.fileBrowser.status.commitState).to.equal(CommitState.CLEAN);
-        expect(container.fileBrowser.status.editorState).to.equal(EditorState.CLEAN);
+        expect(container.fileBrowser.status.commitState).toEqual(CommitState.CLEAN);
+        expect(container.fileBrowser.status.editorState).toEqual(EditorState.CLEAN);
 
         // build output
-        expect(getBuildLogsStub).to.have.been.calledOnce;
-        expect(container.buildOutput.rawBuildLogs.extractErrors(ProgrammingLanguage.JAVA)).to.deep.equal(extractedBuildLogErrors);
-        expect(container.buildOutput.isBuilding).to.be.false;
+        expect(getBuildLogsStub).toHaveBeenCalledTimes(1);
+        expect(container.buildOutput.rawBuildLogs.extractErrors(ProgrammingLanguage.JAVA)).toIncludeSameMembers(extractedBuildLogErrors);
+        expect(container.buildOutput.isBuilding).toBe(false);
 
         // instructions
-        expect(container.instructions).to.exist;
+        expect(container.instructions).not.toBe(null);
```

Comment:
```suggestion
        expect(container.instructions).not.toBe(undefined);
```
I think, since it should be undefined on error. Could you verify this can fail by commenting out whatever sets this?

Reply:
True. I adapted it.